setup.py

```diff
@@ -12,8 +12,24 @@
 from setuptools import setup
 from setuptools.extension import Extension
 
-include_dirs = [os.path.dirname(get_python_inc())]
 library_dirs = [get_config_var("LIBDIR")]
+
+# if the 'libucx' wheel was installed, add its library location to
+# library_dirs so it'll be linked against
+#
+# NOTE: doing this '.__file__' stuff because `sysconfig.get_path("platlib")` and similar
+#       can return paths to the main Python interpreter's installation locations...
+#       not where 'libucx' is going to be installed at build time when using
+#       build isolation
+try:
+    import libucx
+
+    libucx_libdir = os.path.join(os.path.dirname(libucx.__file__), "lib")
+    library_dirs.append(libucx_libdir)
```

Comment:
Try this instead and see if it works
```suggestion
    import libucx
    libs = libucx.load_library()
    libucx_dir = os.path.dirname(libs[0]._name)
    library_dirs.append(libucx_libdir)
```

Reply:
Happy to try this, but for my own understanding, why is it preferable?

And I guess the `try-except` will now also need to catch `OSError` and `RuntimeError` to account for these other conditions: https://github.com/rapidsai/ucx-wheels/blob/ef470edb120771255b5161b29453592073e763b4/python/libucx/libucx/load.py#L37-L52

Reply:
Oh hmm that is a good point about those extra exceptions. A couple of notes:
- I preferred this approach because it makes ucx-py insensitive to the internal file layouts of the ucx wheels. By deferring to the libucx package's `load_library` function, ucx-py doesn't have to know the file layout in libucx.
- I hadn't considered the extra exceptions, and it's making me rethink my suggestion. I was assuming that the libraries would be safely loadable in the environment where we're building, but that may not be true. We want (and need) ucx-py to be possible to build in environments that do not actually have GPUs (like our CI runners), and I don't know if we're guaranteed that libcuda.so will actually load in such environments. It's worth testing out but if this raises exceptions then I don't think they're worth handling because I don't think there's any way to recover from that. If that happens we should probably skip my suggestion here and just stick to your original code.

Reply:
> *because it makes ucx-py insensitive to the internal file layouts of the ucx wheels*

Ahhh yeah great point, I agree that that's desirable.

> *We want (and need) ucx-py to be possible to build in environments that do not actually have GPUs*

Ok yeah I just tried this suggested change without a GPU available, and got one of those errors from `libucx.load_library()`:

```text
RuntimeError: The CUDA driver library libcuda.so.1 was not found on your system.
This library cannot be provided by the libucx wheel and must be installed separately.
```

So I don't think we should use `libucx.load_library()`.

I just pushed a commit with a different approach: https://github.com/rapidsai/ucx-py/pull/1041/commits/0523ba1905d9607962e45a04b5cd751859b3a503

```python
import glob
import libucx

# find 'libucx'
module_dir = os.path.dirname(libucx.__file__)

# find where it stores files like 'libucm.so.0'
libs = glob.glob(f"{module_dir}/**/lib*.so*", recursive=True)

# deduplicate those library paths
lib_dirs = {os.path.dirname(f) for f in libs}
if not lib_dirs:
    raise RuntimeError(
        f"Did not find shared libraries in 'libucx' install location ({module_dir})"
    )

# add all the places 'libucx' stores libraries to that paths
# considered by the linker when compiling extensions
library_dirs.extend(lib_dirs)
```

It's a little more complicated, but not too much, and it avoids assuming anything about the file layout in the `libucx` wheel.

I tested that locally using the approach described in the PR description, everything worked well.

Reply:
cool, thanks for talking through it with me